src/lib/react/jsx/JsxStaticMacro.hx

```diff
@@ -0,0 +1,187 @@
+package react.jsx;
+
+import haxe.macro.Context;
+import haxe.macro.Expr;
+import haxe.macro.Type;
+
+using haxe.macro.Tools;
+
+private typedef JsxStaticDecl = {
+	var module:String;
+	var className:String;
+	var displayName:String;
+	var fieldName:String;
+}
+
+private enum MetaValueType {
+	NoMeta;
+	NoParams(meta:MetadataEntry);
+	WithParams(meta:MetadataEntry, params:Array<Expr>);
+}
+
+class JsxStaticMacro
+{
+	static public var META_NAME = ':jsxStatic';
+
+	static var hasBeenHooked = false;
+	static var decls:Array<JsxStaticDecl> = [];
+
+	static public function injectDisplayNames(type:Expr)
+	{
+		#if !debug
+		return;
+		#end
+
+		// Add hook to generate __init__ at the end of the compilation
+		if (!hasBeenHooked)
+		{
+			hasBeenHooked = true;
+			Context.onAfterTyping(afterTypingHook);
```

Comment:
Does that work with compiler server?

Reply:
Can't find a way to break it, do you have something specific in mind?

```sh
$ haxe --connect 9009 build.hxml --times -debug 
/git/haxe-react/src/lib/react/jsx/JsxStaticMacro.hx:153: afterTyping
/git/haxe-react/src/lib/react/jsx/JsxStaticMacro.hx:153: afterTyping
name           | time(s) |   % |     # | info
----------------------------------------
analyzer       |   0.090 |   9 | 30285 | 
filters        |   0.050 |   5 |     1 | 
generate       |   0.010 |   1 |     1 | 
  js           |   0.010 |   1 |     1 | 
haxelib        |   0.000 |   0 |     1 | 
macro          |   0.550 |  56 |   361 | 
other          |   0.080 |   8 |     1 | 
parsing        |   0.120 |  12 |   647 | 
server         |   0.010 |   1 |  8692 | 
  module cache |   0.010 |   1 |  8692 | 
typing         |   0.080 |   8 |  1116 | 
----------------------------------------
total          |   0.990 | 100 |     0 |

$ cat .build/klabz.js | grep displayName | wc -l
65

$ haxe --connect 9009 build.hxml --times -debug
name             | time(s) |   % |   # | info
----------------------------------------
analyzer         |   0.000 |   0 |  37 | 
filters          |   0.020 |  40 |   1 | 
generate         |   0.020 |  40 |   1 | 
  js             |   0.020 |  40 |   1 | 
haxelib          |   0.000 |   0 |   1 | 
other            |   0.000 |   0 |   1 | 
server           |   0.010 |  20 | 649 | 
  module cache   |   0.010 |  20 | 649 | 
    add modules  |   0.010 |  20 |   7 | 
    changed dirs |   0.000 |   0 | 628 | 
    check        |   0.000 |   0 |   7 | 
typing           |   0.000 |   0 |   1 | 
----------------------------------------
total            |   0.050 | 100 |   0 |

$ cat .build/klabz.js | grep displayName | wc -l
65

$ haxe build.hxml
$ cat .build/klabz.js | grep displayName | wc -l
35

$ haxe build.hxml -debug
/git/haxe-react/src/lib/react/jsx/JsxStaticMacro.hx:153: afterTyping
/git/haxe-react/src/lib/react/jsx/JsxStaticMacro.hx:153: afterTyping

$ cat .build/klabz.js | grep displayName | wc -l
65
```